ApollosProject/apollos-prototype
# Adds Analytics UI Package

## DESCRIPTION

### What does this PR do, or why is it needed?

This extracts all the front end analytics code into an Analytics UI package.

This package does essentially the same thing the baked in code before, with a few improvements. 

- It supports a cleaner interface for calling `track`, either by calling a function and passing in `client` or by using an `AnalyticsProvider`.
- It supports passing in user provided track functions, for integrating easily with third party, client side, analytics libraries. 

### What design trade-offs/decisions were made?

Chose to ditch the `Events` object (referencing hard coded event properties to event names) because I found the structure unwieldy and lacking value at this point.

### How do I test this PR?

1. Clone this PR.
2. In `apolloschurchapp/sc/Providers.js` change line 13 to this 
```
        <AnalyticsProvider trackFunctions={[console.warn]}>
```
3. Navigate throughout the app, doing the following things. 
- Clicking on content items.
- Liking content items
- Sharing content items
- Opening media
- Logging in
- Logging out

and make sure you see the analytics event pop up in a yellow window. You will also see the events flow through the server in the console.


### What automated tests did you write?

Component level tests and unit tests over the store.

## TODO:

- [x] Closes #589 

## REVIEW:

**Manual QA**


**Code Review: Questions to consider**


> The purpose of PR Review is to _improve the quality of the software._
